File: frontend/miniapp/src/api/footprint.ts
import { request } from '@/utils/request'

export interface Footprint {
  id: string
  created_by: string
  name: string
  latitude: number
  longitude: number
  visited_at: string
  note: string | null
  created_at: string
}

export interface FootprintCreate {
  name: string
  latitude: number
  longitude: number
  visited_at: string
  note?: string
}

export interface FootprintUpdate {
  name?: string
  latitude?: number
  longitude?: number
  visited_at?: string
  note?: string
}

export const footprintApi = {
  list: () =>
    request<Footprint[]>({ url: '/api/footprints' }),

  create: (data: FootprintCreate) =>
    request<Footprint>({ url: '/api/footprints', method: 'POST', data }),

  get: (id: string) =>
    request<Footprint>({ url: `/api/footprints/${id}` }),

  update: (id: string, data: FootprintUpdate) =>
    request<Footprint>({ url: `/api/footprints/${id}`, method: 'PUT', data }),

  delete: (id: string) =>
    request<{ message: string }>({ url: `/api/footprints/${id}`, method: 'DELETE' }),
}

File: backend/schemas/footprint.py
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel


class FootprintCreate(BaseModel):
    name: str
    latitude: float
    longitude: float
    visited_at: date
    note: Optional[str] = None


class FootprintUpdate(BaseModel):
    name: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    visited_at: Optional[date] = None
    note: Optional[str] = None


class FootprintOut(BaseModel):
    id: str
    created_by: str
    name: str
    latitude: float
    longitude: float
    visited_at: date
    note: Optional[str]
    created_at: datetime

    class Config:
        from_attributes = True
